### DISCUSSION TITLE
inline search (of tabs)

### DISCUSSION THREAD

**Discussion by Elementari** [mbnuqw/sidebery#340] [Ideas]
Hello, I have moved from Opera to Firefox. In Opera, I have been using v7 tabs, an extension similar to Sideberry. v7 tabs has an extremely helpful feature: It can filter tabs when one enters text into a search field at the top of the extension. One can configure if the search string should search only the Tab's title, or the tab's title AND the URL (I recommend URL and Title). So when one types for example "fire", it will show only tabs that contain "fire" in title or URL. I have been using this feature constantly and I would like to ask if the developer might want to consider adding such a feature to Sideberry. Thank you for considering!!

**Comment by Elementari** [comment]
Hello, I came back wanting to make this suggestion again, only to see I have already posted it. I wanted to re-new the suggestions, it's really the one thing that I'm missing. 
In order to indicate that the tabs are filtered by text, the background color could change at the time when the filter is in effect.
I know there's also the % filter in FF addressbar, but that does not work well for lots of tabs, and also does not have the benefits of a side bar.

Thanks for understanding that I really wanted to repeat my suggestion and thank you very much for Sideberry!!

**Comment by albino1** [comment]
Are you using v4 or v5?  v5 is in technically in beta still, but works very well, and has a search box at the top of the side panel:

https://github.com/mbnuqw/sidebery/releases

v4 isn't likely going to get any more features, so if you want a search box you need to use v5.

**Comment by Elementari** [comment]
Wow, I wasn't aware, thank you very much! 

I am going to wait for the official release of v5 - what great news!

**Reply by albino1** [reply]
Might be a while yet, however v5 is getting almost all of the development attention at this point, so who knows.  Fingers crossed it gets released as stable soon.

**Comment by Elementari** [comment]
Hello, version 5 now does have search (great!), but I was confused about its behavior.
When I search, I get a list of tabs that contain the search word.
But when I click any of these tabs, the search results disappear and the original full tab list is shown. 
I cannot look at each of the found tabs. Instead the search bar is emptied, and for every tab, I have to re-enter the search term!
So when there are 10 results for search word "headphone", I will have to type "headphone" 10 times in order to look at each of these results, it seems. 

I would have thought that for example cliking the magnifying glass on the left hand would toggle between showing the search results (and showing the text "headphone" in the search field), and NOT showing that text,but showing all tabs (and clicking again would show the last search results again.

**Reply by macgyverkam** [reply]
It should be more like filtering than searching

**Reply by Elementari** [reply]
Apart from an option to make the search results stay after clicking a tab, I would also really apreciate a feature that will only HIGHLIGHT tabs (with a background color) that match the search, rather than filtering for them. This would be super useful. There should also be an option that would cause this color to be applied also when filtering.

So, it could be like this:

In Settings, these options:

( ) Highlight search result tabs with background color (choose color)
 --- (  ) Don't highlight when filtering search results (only hights when all tabs remain visible)

( ) Highlight part of text that corresponds to search text (choose color)
  ---  (  ) Don't highlight when filtering search results (only hights when all tabs remain visible)

( ) Toggling between search highlighting and search filtering retains search text in search field


In sidebar, there would be two icons:
* magnifying glass for filtering search (just like it already exists now; could be replaced by filter symbol)
--- (when filter is active, icon changes background color (choose color))
* a different icon of developer's choice: causes search highligting (all tabs remain visible, but tabs and/or search words are highlighted with color).

**Reply by DWolf19** [reply]
> an option to make the search results stay after clicking a tab

This please
